### PR TITLE
add json and medium/long text column type

### PR DESCRIPTION
**Pull request by matthewjumpsoffbuildings**
Json column type should be supported in the Schema Grammar.

Also added `typeMediumText()` and `typeLongText()` methods, no reason to not just map them to the basic `typeText()`

### CODE REVIEW

**Comment by taka-oyama** [comment]
Add tests please.

File: src/Schema/Grammar.php [hunk 1]
```diff
@@ -403,6 +403,39 @@ protected function typeText(Fluent $column)
         return "string(max)";
     }
 
+    /**
+     * Create the column definition for a medium text type.
+     *
+     * @param  \Illuminate\Support\Fluent  $column
```
Comment (taka-oyama) [inline src/Schema/Grammar.php:409]:
```suggestion
     * @param  Fluent<string, mixed> $column
```

File: src/Schema/Grammar.php [hunk 2]
```diff
@@ -403,6 +403,39 @@ protected function typeText(Fluent $column)
...
+     */
+    protected function typeMediumText(Fluent $column)
+    {
+        return $this->typeText($column);
+    }
+
+    /**
+     * Create the column definition for a long text type.
+     *
+     * @param  \Illuminate\Support\Fluent  $column
```
Comment (taka-oyama) [inline src/Schema/Grammar.php:420]:
```suggestion
     * @param  Fluent<string, mixed> $column
```

File: src/Schema/Grammar.php [hunk 3]
```diff
@@ -403,6 +403,39 @@ protected function typeText(Fluent $column)
...
+     */
+    protected function typeLongText(Fluent $column)
+    {
+        return $this->typeText($column);
+    }
+
+    /**
+     * Create the column definition for a json type.
+     *
+     * @param  \Illuminate\Support\Fluent  $column
```
Comment (taka-oyama) [inline src/Schema/Grammar.php:431]:
```suggestion
     * @param  Fluent<string, mixed> $column
```

**Review by taka-oyama — CHANGES_REQUESTED** [review]

**Comment by matthewjumpsoffbuildings** [comment]
tests added, hopefully that is sufficient?

File: tests/Schema/BlueprintTest.php [hunk 4]
```diff
@@ -476,6 +488,7 @@ public function test_default_values(): void
         self::assertSame(0.1, $result['float']);
         self::assertSame(true, $result['bool']);
         self::assertSame('a', $result['string']);
+        self::assertSame('[1,2,3]', $result['json']);
```
Comment (taka-oyama) [inline tests/Schema/BlueprintTest.php:491]:
```suggestion
        self::assertSame('a', $result['string_max']);
        self::assertSame('a', $result['medium_text']);
        self::assertSame('a', $result['long_text']);
        self::assertSame('[1,2,3]', $result['json']);
```

**Comment by taka-oyama** [comment]
Yes. Thank you.

**Pull request merged**